practice/rain.py

```diff
@@ -4,139 +4,99 @@
 # 1. Setup
 import operator
 
+
+# 2. Define
+
 def import_table():
-    with open('sunnysiderain.txt', 'r') as rain_table:
-        initial_list = rain_table.readlines()
-    return initial_list
+    """This imports the table to be read from the file"""
+    with open('sunnysiderain.txt') as rain_table:
+        intial_table = rain_table.readlines()
+    stripped_table = strip_table(intial_table)
+    return cut_out_items(stripped_table)
 
 def strip_table(table):
-    r"""
+    r"""Strips the table of formatting.
+
     >>> strip_table([''] * 12 + ['A \n'])
     [['A']]
-
-    :param table:
-    :return:
     """
-    stripped_table = [line.strip().split() for line in table[12:]]
-    return stripped_table
+    return [line.strip().split() for line in table[12:]]
 
 def cut_out_items(stripped_table):
-    """
+    """Cuts out unused parts of the imported data.
+
     >>> cut_out_items([['A', 'B', 'C']])
     [['A', 'B']]
-
-    :param stripped_table:
-    :return:
     """
-    cut_table = [line[:2] for line in stripped_table]
-    return cut_table
+    return [line[:2] for line in stripped_table]
 
-def make_lookup_table(formatted_table):
-    """
-    >>> make_lookup_table([['A', 'B']])
-    {'A': 'B'}
+def get_date_to_amt(table):
+    """Turns data into a dictionary that pairs each date with the rainfall amount on that day.
 
-    :param formatted_table:
-    :return:
+    >>> get_date_to_amt([['A', '3']])
+    {'A': 3}
     """
-    lookup_table = {key: value for (key, value) in formatted_table}
-    return lookup_table
+    date_to_amt = {key: value for (key, value) in table}
+    return filter_table(date_to_amt)
+
+def find_date_with_most_rain(date_to_amt):
+    """Compares all dictionary items and returns the date with the highest value.
 
+    >>> find_date_with_most_rain({'A': 1, 'B': 2})
+    'B'
+    """
+    return max(date_to_amt.items(), key=operator.itemgetter(1))[0]
 
+def filter_table(date_to_amt):
+    """Removes dates which have no data from data set.
 
-def find_date_with_most_rain(filtered_table):
+    >>> filter_table({'15-DEC-2010': '-', '24-JUL-2009': '0'})
+    {'24-JUL-2009': 0}
     """
-    >>> find_date_with_most_rain({'A': '1', 'B': '2'})
-    B
+    return {date: int(amt) for date, amt in date_to_amt.items() if not '-' == amt}
```

Comment:
Great work!

My only suggestion is to avoid "yoda expressions". Put variables first and constants second in if expressions.

Instead of

```py
if not '-' == act
```

Put the var first

```py
if not act == '-'
```